com.unity.shadergraph/Editor/ShaderFoundry/Utilities/KeywordAttribute.cs

```
@@ -0,0 +1,212 @@
+            return builder.Build();
+        }
+
+        internal ShaderAttribute Build(ShaderContainer container)
+        {
+            var attributeBuilder = new ShaderAttribute.Builder(container, AttributeName);
+            if (KeywordName != null)
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, KeywordNameParamName, KeywordName).Build());
+            if (Scope != null)
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, ScopeParamName, Scope).Build());
+            if (Stage != null)
+                attributeBuilder.Param(new ShaderAttributeParam.Builder(container, StageParamName, Stage).Build());
+            attributeBuilder.Param(new ShaderAttributeParam.Builder(container, KeywordModeParamName, KeywordMode.ToString()).Build());
+            return attributeBuilder.Build();
+        }
+
+        internal static BoolKeywordAttribute FindFirst(IEnumerable<ShaderAttribute> attributes)
+        {
+            var attribute = attributes.FindFirst(AttributeName);
+            if (attribute.IsValid)
+                return TryParse(attribute);
+            return null;
+        }
+
+        internal static BoolKeywordAttribute TryParse(ShaderAttribute attribute)
+        {
+            if (attribute.Name != AttributeName)
+                return null;
+            var result = new BoolKeywordAttribute();
+            foreach (var param in attribute.Parameters)
+            {
+                if (param.Name == KeywordNameParamName)
+                    result.KeywordName = param.Value;
+                else if (param.Name == KeywordModeParamName)
+                    Enum.TryParse(param.Value, out result.KeywordMode);
+                else if (param.Name == ScopeParamName)
+                    result.Scope = param.Value;
+                else if (param.Name == StageParamName)
+                    result.Stage = param.Value;
```

Looks like we silently ignore anything that doesn't match what we expect.

---

At the moment yes. I would like to unify attribute parsing a bit more as it's more nuanced due to positional/named argument handling.